### ISSUE TITLE
[ISSUE] Cracked Whitelisting Not Possible

### ISSUE THREAD

**Issue by ytcalifax** [Jakkoble/TwitchWhitelist#3]
The whitelisted UUID is always the premium UUID of the player - however, this won't work for cracked servers (in my country most players are cracked), as the cracked UUID is different. A possible way to solve this issue is to make it so the player gets whitelisted upon attempting to join with the name (as then you would possibly be able to fetch the UUID they are joining with).

Another (and possible easier/better) way to do this would be to whitelist by username and not UUID, since the names of players are unique just like the UUIDs.

**Comment by ytcalifax** [comment]
Another way to do this would be to whitelist by username and not UUID.

**Comment by Jakkoble** [comment]
Hey, thank you for your thoughts!

The problem is the Following: If I whitelist the Players via their PlayerNames instead of the UUID, you could easily join with an OfflineAccount in a cracked Launcher. There you can play and join with every PlayerName you want on offline Servers. So these players can easily **bypass** the Whitelist and this would be a potential **securtiy risk** to your server. E. g. Your Minecraft Name is "ytcalifax" and if you are offline, another Player joins with an Offline Account with the Name "ytcalifax" on the Server and bypassed the Whitelist. 

To sum up, whitelisting on offline Servers is a pretty weird thing and I don't think that there is any good solution to fix this issue so the plugin is for only for Online Servers. :/

_The Vanilla Minecraft whitelist has also not fixed this issue yet so Whitelist on offline Servers is useless._

**Comment by ytcalifax** [comment]
You could add a switch (in the config) that allows one to use the plugin on cracked servers - this way cracked servers can use it and the online ones could use it too, as **premium servers can use the UUID check**  and **cracked ones the name check**.

This way if one (like me) wants to use it to stop non-whitelisted players from registering an account (via AuthMe) it would be possible.

**Comment by Jakkoble** [comment]
I really understand your needs in this situation and I appreciate your interrests in my Plugin.

But I don't know if you understood my right, **whitelisting with PlayerNames** on Offline Servers is pretty **useless**. Every Player can easily join with whatever PlayerName they want. It takes around 5 minutes to bypass the Whitelist. So every Player (also non-whitelisted Players) are able to join if they want. 

So checking by UserName will not work and the whitelist in general is pointless.

**Comment by ytcalifax** [comment]
I did understand you, however you seem to have misunderstood me - I am using TTV-WL not by itself, but alongside AuthMe (i.e if they do join with **ytcalifax** for example, **they would have to enter a password**). In my use case I am using the plugin **just to stop players that have not redeemed points from being able to even join in order to do the /register  command**.

**Comment by Jakkoble** [comment]
Oh well, now I know what you meen. So I am trying to implement your needs right now. I will be contacting you within today.

**Comment by ytcalifax** [comment]
Thanks! I'll make sure to leave an informative 5-star review once this gets solved 😁

**Comment by Jakkoble** [comment]
Hey there!

I just fixed your Issue and I hope everything is working fine for you now. You can download the new Plugin Version **[here](https://github.com/Jakkoble/TwitchWhitelist/releases/latest)**.

If you need any additional help, have other feedback or solution feel free to write back or open a new Issue.

**Comment by ytcalifax** [comment]
Thanks! I left you a nice review on SpigotMC 😁. Keep on coding awesome plugins.
